src/OptiHPLCHandler/empower_methodset_method.py

```diff
@@ -21,32 +21,28 @@ def __init__(self, method_definition: Union[dict, list]):
         self.column_oven_list: list[ColumnHandler] = []
         self.instrument_method_list: list[InstrumentMethod] = []
 
-        if isinstance(method_definition, dict):
+        if isinstance(method_definition, dict) and "results" in method_definition:
             # If the entire response from Empower is passed, extract the results
-            method_definition = method_definition["results"]
-        for instrument_method_definition in method_definition:
+            method_definition = method_definition["results"][0]
+        self.original_method = method_definition
+        for instrument_method_definition in method_definition["modules"]:
             instrument_method = instrument_method_factory(instrument_method_definition)
             self.instrument_method_list.append(instrument_method)
             if isinstance(instrument_method, ColumnHandler):
                 self.column_oven_list.append(instrument_method)
 
-    @property
-    def original_method(self):
-        """
-        Return the original method definition.
-
-        :return: The original method definition.
-        """
-        return [method.original_method for method in self.instrument_method_list]
-
     @property
     def current_method(self):
         """
         Return the current method definition.
 
         :return: The current method definition.
         """
-        return [method.current_method for method in self.instrument_method_list]
+        method = self.original_method.copy()
```

Comment:
Deep copy needed?

Reply:
I don't want to change the `"modules"` value of the original method, which is why I copy it.

I probably should make self.original_method into an immutable mapping, to make it clear that it is not to be changed, and to avoid doing it by mistake.

Reply:
Ah, sorry, I misunderstood you. Yes, I need to use deepcopy.

Reply:
So, I don't have to make a deep copy, since I don't mutate the underlying objects - I extract a string and change that, but strings aren't mutable, so that creates a new string. Let's discuss whether we need to it anyway tomorrow.